### PR TITLE
CBG-2058: Always pass a terminator to Compact

### PR DESCRIPTION
**Pull request by markspolakovs**
CBG-2058

If SGW was left running long enough that the compaction background task runs (so only on nodes with import enabled), and there are tombstones present older than the metadata purge interval, the task would check that a nil terminator is complete and thus panic. This PR ensures we always pass down a terminator to Compact, as well as adding a regression test.

## [Integration Tests](https://jenkins.sgwdev.com/job/SyncGateway-Integration/build?delay=0sec)
- [x] `xattrs=true` https://jenkins.sgwdev.com/job/SyncGateway-Integration/249/

### CODE REVIEW

File: db/database_test.go [hunk 1]
```diff
@@ -66,6 +66,11 @@ func setupTestDBForBucketWithOptions(t testing.TB, tBucket base.Bucket, dbcOptio
 
 func setupTestDBWithOptionsAndImport(t testing.TB, dbcOptions DatabaseContextOptions) *Database {
 	AddOptionsFromEnvironmentVariables(&dbcOptions)
+	if dbcOptions.GroupID == "" && base.IsEnterpriseEdition() {
+		// In a real SG this would be set in DefaultStartupConfig.
+		dbcOptions.GroupID = "default"
+		RegisterImportPindexImpl("default")
```
Comment (bbrks) [inline db/database_test.go]: This seems odd that it's not already done as part of `NewDatabaseContext` in the real SG code.
Having it done inside the function that generates an options struct (`dbcOptionsFromConfig`) feels a bit like a side-effect.

File: db/database_test.go [hunk 2]
```diff
@@ -66,6 +66,11 @@ func setupTestDBForBucketWithOptions(t testing.TB, tBucket base.Bucket, dbcOptio
 
 func setupTestDBWithOptionsAndImport(t testing.TB, dbcOptions DatabaseContextOptions) *Database {
 	AddOptionsFromEnvironmentVariables(&dbcOptions)
+	if dbcOptions.GroupID == "" && base.IsEnterpriseEdition() {
+		// In a real SG this would be set in DefaultStartupConfig.
+		dbcOptions.GroupID = "default"
```
Comment (bbrks) [inline db/database_test.go]: `"default"` is actually not the default beyond the SG config file, it's an empty string - but it not working with empty strings implies there might be a larger problem here...

**Review by bbrks — CHANGES_REQUESTED** [review]
Have a few questions about the group ID stuff - maybe we can walk through the changes in a call to discuss instead of back-and-forth

File: db/database_test.go (hunk 1 above)
Comment (markspolakovs) [reply db/database_test.go]: Discussed offline - this should indeed be moved into `NewDatabaseContext`, but that's slightly outside the scope of this PR (and it's risky for a PR that's getting backported to 3.0.1), so I'll leave this as is with a TODO (to get the tests to pass) and file a separate one to move this out.

**Review by bbrks — APPROVED** [review]
LGTM

**Pull request merged**